### PR TITLE
Sign registration transaction

### PR DESCRIPTION
**Pull request by thestrabusiness**
This commit adds functionality to the registration modal that allows an
unregistered user to sign an EIP712-typed payload to permit DAI transfer
and register for the course.

The functionality to sign a transaction is wrapped in a hook that takes
a provider and the user's address. The hook returns a function that
takes the users nonce for the DAI contract and an expiry time for the
permit transaction. Upon completion of signing, the function returns an
object with properties v, r and s, to pass into the transaction.

### CODE REVIEW

File: src/course/course.js [hunk 1]
```diff
@@ -23,3 +28,37 @@ export const isRegistered = async (learner, provider) => {
...
+      nonce,
+      expiry,
+      v,
+      r,
+      s,
+      {gasLimit: 200000}
+    ));
+  } catch (err) {
+    // Handle error
+    // console.log(err);
```
Comment (andytudhope) [inline src/course/course.js:61]: We should handle errors elegantly and make sure we let learners know if their transaction has encountered an unexpected road bump and what to do (i,e. try again later).

**Review by probablyangg — APPROVED** [review]

**Comment by probablyangg** [comment]
Haven’t tested it, but the code for signing looks good

I think next step would be to merge the two PRs we have. Let’s definitely sync on it

**Comment by andytudhope** [comment]
I'll leave this open for now as you play around with integrating your work on signing and modal fixes with Angela's more global approach to things.

We'll all look forward to another PR on the `card` component illustrating how we might refactor the code for greater readability and ease of contribution, as well as any thoughts you have on how to streamline collaboration.

Thank you so, so much :heart: :pray:

File: src/course/course.js (hunk 1 above)
Comment (thestrabusiness) [reply src/course/course.js:61]: Agreed! It looks like we might be able to cook up a notification component that we can use in this case and others. I'll play around with that next.

**Pull request merged**